src/main/webapp/src/components/songInfo.js

```diff
@@ -94,18 +100,53 @@ class SongInfo extends React.Component {
     });
   }
 
+  /** Save state of the component to localStorage. */
+  saveState() {
+    const json = JSON.stringify(this.state);
+    localStorage.setItem("state", json);
+  }
+
+  /** Checks if the state of the component was loaded successfully. */
+  isReady(componentState) {
+      return (componentState && componentState.isLoading === false && componentState.errorMsg === null);
```

Comment:
wouldn't `!componentState.isLoading` instead of `componentState.isLoading === false` work?

Reply:
hmmm, yes it will work, it this method I was just expecting that state object will definitely have `isLoading` and `errorMsg` fields, that's why I made this check more concrete, but now I see that it's not necessary because of te default state of components, so I'll change it